### DISCUSSION TITLE
Trying to create a Control Library... Runtime errors.

### DISCUSSION THREAD

**Discussion by LaurentInSeattle** [AvaloniaUI/Avalonia#15648] [Q&A]
Hi all, 

I am trying to create a custom library of controls in my training project, here: https://github.com/LaurentInSeattle/AvaloniaExperiments 

I have been able to create two custom UserControls in the library without major troubles. 
(SvgIcon, BadgeControl both unfinished in the above project, not looking so good, but working) 

So then I tried to create a "lookless" control, named just "Badge", in the same project, based on the code from the AuraUI project.
The CSharp for the control is in the Control library and the styles are in a separate "Theme" assembly.
All that does compiles but loading of the XAML for the style of the control fails at run time, complaining it cannot access the Badge class.

Any idea of what I have done wrong ? 

Thanks in advance.

**Comment by timunie** [comment]
I suggest to base on `TemplatedControl` or `ContentControl` or similar, based on your needs. A `UserControl` is more meant to be used as a "Page" (Name is confusing, it was in WPF already ^^). 

Sample: https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/CustomControls/RatingControlSample
